### DISCUSSION TITLE
Switch between simulation and real environment

### DISCUSSION THREAD

**Discussion by saud-2** [lavaloon-eg/ksa_compliance#61] [General]
Hello,
How to Switch between simulation and real environment or the opposite? and get production CSID for real environment or for both ?
regards.

**Comment by mhaggag** [comment]
Hello.

Currently, the environment is controlled by the Fatoora server URL in the "Integration" tab of `ZATCA Business Settings`. Generally, we do not switch environments on the same ERPNext instance--invoices sent to one environment will not be sent to another environment. They'll be marked as reported (or rejected etc., just no longer ready for sending to ZATCA).

Ideally you create a UAT or testing instance (cloned from or configured similarly to production), and test the operation using sandbox or simulation. Then you apply the production configuration on the production environment and onboard, perform compliance, and go live.

Regardless, to switch the environment on the same instance:
* Change the URL and save
* Onboard to replace the CSR and get a compliance request ID
* Perform compliance checks
* Get production CSID to replace the current one if any
